test/SendCmioResponse.t.sol

```diff
@@ -0,0 +1,177 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+pragma solidity ^0.8.0;
+
+import "forge-std/StdJson.sol";
+import "forge-std/console.sol";
+import "forge-std/Test.sol";
+
+import "src/EmulatorConstants.sol";
+import "src/SendCmioResponse.sol";
+import "./BufferAux.sol";
+
+contract SendCmioResponse_Test is Test {
+    using Buffer for Buffer.Context;
+    using BufferAux for Buffer.Context;
+    using stdJson for string;
+
+    // configure the tests
+    string constant JSON_PATH = "./test/uarch-log/";
+    string constant CATALOG_PATH = "catalog.json";
+    string constant SEND_CMIO_RESPONSE_PATH = "send-cmio-response-steps.json";
+
+    uint256 constant siblingsLength = 59;
+
+    struct Entry {
+        string binaryFilename;
+        string finalRootHash;
+        string initialRootHash;
+        string logFilename;
+        bool proof;
+        uint256 proofsFrequency;
+        uint256 steps;
+    }
+
+    struct RawAccess {
+        uint256 addressAccess;
+        uint256 log2_size;
+        string read_hash;
+        string read_value;
+        string[] sibling_hashes;
+        string typeAccess;
+        string written_hash;
+    }
+
+    function testSendCmioResponse() public {
+        Entry[] memory catalog =
+            loadCatalog(string.concat(JSON_PATH, CATALOG_PATH));
+        string memory resetLog =
+            string.concat(JSON_PATH, SEND_CMIO_RESPONSE_PATH);
+
+        // all tests combined can easily run out of gas, stop metering
+        // also raise memory_limit in foundry.toml per https://github.com/foundry-rs/foundry/issues/3971
+        vm.pauseGasMetering();
+        // create a large buffer and reuse it
+        bytes memory buffer = new bytes(100 * (siblingsLength + 1) * 32);
+
+        for (uint256 i = 0; i < catalog.length; i++) {
+            if (
+                keccak256(abi.encodePacked(catalog[i].logFilename))
+                    != keccak256(abi.encodePacked("send-cmio-response-steps.json"))
+            ) {
+                continue;
+            }
+            console.log("Replaying log file %s ...", catalog[i].logFilename);
+            require(
+                catalog[i].proofsFrequency == 1, "require proof in every step"
+            );
+
+            string memory rj = loadJsonLog(resetLog);
+
+            bytes32 initialRootHash =
+                vm.parseBytes32(string.concat("0x", catalog[i].initialRootHash));
+            bytes32 finalRootHash =
+                vm.parseBytes32(string.concat("0x", catalog[i].finalRootHash));
+
+            loadBufferFromRawJson(buffer, rj);
+
+            AccessLogs.Context memory accessLogs =
+                AccessLogs.Context(initialRootHash, Buffer.Context(buffer, 0));
+
+            // Prepare arguments for sendCmioResponse
+            // These values are hard-coded in order to match the values used when generating the test log file
+            uint16 reason = 1;
```

Comment:
I don't understand this. I checked the value of `HTIF_YIELD_REASON_ADVANCE_STATE` in cpp code, and it's "0". Why do we use 1 here?

Reply:
The reason parameter accepts any value.
This [lua script](https://github.com/cartesi/machine-emulator/blob/main/tests/lua/uarch-riscv-tests.lua#L561) sets reason = 1 to enable testing of bit shifts used to construct the htif-fromhost value.

Reply:
So the logs for testing `send-cmio-response` are actually generated by using `HTIF_YIELD_REASON_INSPECT_STATE` which has value "1", right?